pkg/graph/constructor.go

```diff
@@ -94,3 +96,53 @@ func (g *Graph) AddChannel(channel messagingv1.Channel) {
 
 	v.AddEdge(to, dest, NoTransform)
 }
+
+func (g *Graph) AddTrigger(trigger eventingv1.Trigger) error {
+	brokerRef := &duckv1.KReference{
+		Name:       trigger.Spec.Broker,
+		Namespace:  trigger.Namespace,
+		APIVersion: "eventing.knative.dev/v1",
+		Kind:       "Broker",
+	}
+	brokerDest := &duckv1.Destination{Ref: brokerRef}
+	broker, ok := g.vertices[makeComparableDestination(brokerDest)]
+	if !ok {
+		return fmt.Errorf("trigger refers to a non existent broker, can't add it to the graph")
+	}
+
+	triggerRef := &duckv1.KReference{
+		Name:       trigger.Name,
+		Namespace:  trigger.Namespace,
+		APIVersion: "eventing.knative.dev/v1",
+		Kind:       "Trigger",
+	}
+	triggerDest := &duckv1.Destination{Ref: triggerRef}
+
+	to, ok := g.vertices[makeComparableDestination(&trigger.Spec.Subscriber)]
+	if !ok {
+		to = &Vertex{
+			self: &trigger.Spec.Subscriber,
```

Comment:
Here if the subscriber doesn't exist in the form of a graph node, we will be creating one for them. But for the brokerDest comment above, we will just return the error. Shouldn't we do the same if we don't have the mechanism to ensure the destination broker is always created first

Reply:
No, the subscriber may be something like an arbitrary URL (which we wouldn't be able to check for).

The reason we need the broker to be created first is that otherwise we won't be able to correctly create the broker with all required config and edges. Here, instead we are creating the subscriber if it does not exist because there is no way for us to guarantee that we created all the subscribers as nodes beforehand (we cant list arbitrary URLS)